compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -511,44 +511,20 @@ bool is_const(const loco::Node *node)
   return true;
 }
 
-// NOTE Following conditions can be extended later
-// NOTE Used for Maximum, Miminum as ReLU/ReLU6
-//
-// Find T with an NCHW pattern described below
-//   - Input (non-constant) shape : [N, C, H, W]
-//   - Input (constant) shape : [1] or []
-//   - Output shape : [N, C, H, W]
-template <class T>
-bool is_NCHW_with_s_const(const T *node, luci::CircleNode *&pred_node,
-                          luci::CircleConst *&comp_const)
+bool is_scalar_const(const loco::Node *node)
 {
-  auto x = dynamic_cast<luci::CircleConst *>(node->x());
-  auto y = dynamic_cast<luci::CircleConst *>(node->y());
-
-  if (x != nullptr && y == nullptr)
-  {
-    pred_node = loco::must_cast<luci::CircleNode *>(node->y());
-    comp_const = x;
-  }
-  else if (x == nullptr && y != nullptr)
-  {
-    pred_node = loco::must_cast<luci::CircleNode *>(node->x());
-    comp_const = y;
-  }
-  else
-  {
-    // Ignore if T does not have a comp_const input.
+  auto const_node = dynamic_cast<const luci::CircleConst *>(node);
+  if (not const_node)
     return false;
-  }
 
-  if (pred_node->rank() != 4)
+  const auto const_rank = const_node->rank();
+  // shape of scalar
+  // 1. rank = 0
+  // 2. rank = 1, dimension = 1
+  if (const_rank != 0 and (const_rank == 1 && const_node->dim(0).value() == 1))
```

Comment:
Maybe?
```suggestion
  if (const_rank != 0 and (const_rank != 1 || const_node->dim(0).value() != 1))
```
(optional)
IMHO, check if true condition seems easier to read. I mean original code looks better.

Reply:
And even though condition is wrong, all the tests are failed. Related test might be needed.

Reply:
You're right. I've updated conditions and added tests. PTAL